hw_diag/utilities/miner.py

```diff
@@ -8,9 +9,21 @@ def fetch_miner_data(diagnostics):
     try:
         peerbook = client.get_peer_book()[0]
         height = client.get_height()
+        # Used for unit test to make sure exception
+        # can be thrown from hm-pyhelper
+        if "MC" in diagnostics and diagnostics['MC'] == 'testexception':
+            # Raise exception unit test error
+            raise MinerFailedFetchData("Unable to fetch miner data "
+                                       "from miner container "
+                                       "via JSON RPC API. Exception: %s" %
+                                       "Exception Test Error")
     except Exception as err:
-        raise Exception("Unable to fetch miner peerbook from miner container "
-                        "via JSON RPC API. Exception: %s" % str(err))
+        # Raise exception as MinerFailedFetchData exception from hm-pyhelper
+        raise MinerFailedFetchData("Unable to fetch miner data "
```

Comment:
This change needs to be paired with some logic in [tasks.py](https://github.com/NebraLtd/hm-diag/blob/12420cfb08880a0b296b4130174f8563c8074ea1/hw_diag/tasks.py#L42) so that the exception is handled and doesn't prevent a report from being generated.